index.rst

```diff
@@ -2,24 +2,180 @@
+They are used in parallel with SAL Scripts to execute tasks that require some user interaction or a finer degree of control over the operations.
+
+Although they can be very useful in the early development stages of the process, we do not recommend storing any long-term use software in notebooks.
+The main reason is that it is currently note possible to unit test Jupyter notebooks, which can leave to bit rot in short timescales, especially with the rapidly evolving software development one would expect in early commissioning stages.
+
+Exceptional Cases
+-----------------
+
+.. Since part of Telescope and Site Software scope is to deliver a CI/CD system for the observatory control software, this system
+
+Although a substantial number of requirements can be verified with the use of our CI system, we acknowledge that some of them will inevitably demand a more individualized approach and, sometimes, even a more involved process demanding additional data analysis and/or a more descriptive verification methodology.
+
+Ideally, these cases would be addressed by either developing a customized SAL Script to drive the system through the required test steps or, in the more extreme conditions that might require interactive steps, a Jupyter notebook.
+
+Furthermore, any detailed data analysis/methodology should be included in a technical note to be incorporated into the requirement verification artifacts.
+
+.. , since we execute (or plan to) these tests continuously, we can verify that the system continues to fullfil requirements at every iteration.
+
+.. Because these functionalities are (mostly) taken from the requirements pool, it means that each new feature is generally accompanied by a requirement verification test.
+.. Nevertheless, tests comes in a variety of different shapes and forms and, most importantly, sometimes it is not possible to fully exercise a given requirement in a testing environment.
+
+.. _verification-artifacts:
+
+Verification artifacts
+======================
+
+In general, successfully executing the testing suite from start to finish gives us confidence that the system is operating as expected.
+Nevertheless, the SAL Scripts used during these tests are only designed to fail execution if some critical condition happen (e.g. a component unexpectedly went to FAULT state), and are not designed to verify all the details of every single component involved in the process.
+
+For example, the ``set_summary_state`` SAL Script is designed to execute state transitions and is commonly used in operations to enable components, recover from fault states, etc.
+They are executed as an initial step during testing to enable all components and, while they verify the components do transition successfully to the desired state, they **do not** check that the CSCs publish all the required events.
+This type of verification need to be done a posteriori and included in verification artifacts.
```

Comment:
but it is, right? This is testing that Michael does...

Also... not sure that it's relevant, but that's LSE-209...which is technically for SE to verify....

Reply:
> but it is, right? This is testing that Michael does...

Yes this is being done, which is kind of the point.. Since we are already doing it, we should use these tests artifacts as verification artifacts.

> Also... not sure that it's relevant, but that's LSE-209...which is technically for SE to verify....

So, I am not distancing SE level requirements from TSSW level requirements. The idea is to provide a global solution for the observatory control system software. In the end it will flow down from SE to TSSW anyway.